x-pack/plugins/infra/public/components/asset_details/hooks/use_asset_details_state.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import createContainer from 'constate';
+import { useMemo } from 'react';
+import DateMath from '@kbn/datemath';
+import type { AssetDetailsProps } from '../types';
+
+const DEFAULT_DATE_RANGE = {
+  from: 'now-15m',
+  to: 'now',
+};
+
+export interface UseAssetDetailsStateProps {
+  state: Pick<
+    AssetDetailsProps,
+    'node' | 'nodeType' | 'overrides' | 'dateRange' | 'onTabsStateChange'
+  >;
+}
+
+export function useAssetDetailsState({ state }: UseAssetDetailsStateProps) {
+  const { node, nodeType, dateRange: rawDateRange, onTabsStateChange, overrides } = state;
+
+  const dateRange = useMemo(() => {
```

Comment:
As this is similar to `getParsedDateRange` inside `use_unified_search.ts` does it make sense to extract it as a helper and reuse it 🤔

Reply:
Good point. I think it makes sense. Will do that.

Reply:
Actually, we'll be moving the asset_details to another plugin at some point. Now I'm not sure if it's worth the effort to do this now.